legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/project/ProjectStructure.java

```diff
@@ -430,14 +436,40 @@ else if (requireRevisionId)
+        {
+            oldProjectType = ProjectType.MANAGED;
+            if (updateBuilder.configUpdater.getProjectType() == null)
+            {
+                updateBuilder.configUpdater.setProjectType(oldProjectType);
+            }
+        }
+        ProjectType newProjectType = updateBuilder.configUpdater.getProjectType() == null ? oldProjectType : updateBuilder.configUpdater.getProjectType();
+        // For MANAGED projects
+        if (updateBuilder.projectStructureExtensionProvider != null && newProjectType != ProjectType.EMBEDDED)
+        {
+            // converted from EMBEDDED set version
+            if (oldProjectType == ProjectType.EMBEDDED && updateBuilder.configUpdater.getProjectStructureVersion() == null)
+            {
+                updateBuilder.configUpdater.setProjectStructureVersion(currentConfig.getProjectStructureVersion().getVersion());
+            }
+            // if extension version was not specified, use the latest one for given version
+            if (updateBuilder.configUpdater.getProjectStructureVersion() != null && updateBuilder.configUpdater.getProjectStructureExtensionVersion() == null)
+            {
+                updateBuilder.configUpdater.setProjectStructureExtensionVersion(updateBuilder.projectStructureExtensionProvider.getLatestVersionForProjectStructureVersion(updateBuilder.configUpdater.getProjectStructureVersion()));
+            }
+        }
         ProjectConfiguration newConfig = updateLegacyDependencies(updateBuilder.getProjectConfigurationUpdater().update(currentConfig), projectFileAccessProvider);
 
         validateProjectConfiguration(newConfig);
 
-        // prevent downgrading project
-        if (newConfig.getProjectStructureVersion().compareTo(currentConfig.getProjectStructureVersion()) < 0)
+        // prevent downgrading project or using extensions on non-managed projects
+        if (newConfig.getProjectType() != ProjectType.EMBEDDED && newConfig.getProjectStructureVersion().compareTo(currentConfig.getProjectStructureVersion()) < 0
+                || newConfig.getProjectType() == ProjectType.EMBEDDED && (newConfig.getProjectStructureVersion().getVersion() < currentConfig.getProjectStructureVersion().getVersion() || newConfig.getProjectStructureVersion().getExtensionVersion() != null))
         {
-            throw new LegendSDLCServerException("Cannot change project " + projectId + " from project structure version " + currentConfig.getProjectStructureVersion().toVersionString() + " to version " + newConfig.getProjectStructureVersion().toVersionString(), Status.BAD_REQUEST);
+            throw new LegendSDLCServerException("Cannot change project " + projectId + " from project " + currentConfig.getProjectType() + " structure version " + currentConfig.getProjectStructureVersion().toVersionString() + " to " + newConfig.getProjectType() + " version " + newConfig.getProjectStructureVersion().toVersionString(), Status.BAD_REQUEST);
```

Comment:
Let's separate this into two error conditions. The one that was previously there, preventing downgrading, should remain as it was before. It should apply to both embedded and managed projects. The second condition should apply only to embedded project structures where there is an attempt to set the extension version.

Reply:
done, however for project converted from MANAGED to EMBEDDED had to take in account that existing extension version need to be omitted when comparing.